ir/function.cpp

```diff
@@ -293,12 +362,129 @@ bool Function::removeUnusedStuff(const multimap<Value*, Value*> &users,
   return changed;
 }
 
+static string_view bb_dot_name(const string &name) {
+  if (name[0] == '%')
+    return string_view(name).substr(1);
+  return name;
+}
+
 void Function::unroll(unsigned k) {
   if (k == 0)
     return;
   LoopAnalysis la(*this);
   ofstream out("a.gv");
   la.printDot(out);
+
+  auto &Nodes = la.getHeaderNodes();
+  const auto &[roots, bottomup, topdown] = la.getNestedTree();
+
+  BasicBlock &sink = getBB("#sink");
+
+  auto Type = la.getNodeType();
+
+  vector<pair<const BasicBlock*, bool>> worklist;
+
+  for (auto &root : roots) {
+    worklist.emplace_back(root, false);
+  }
+
+  while (!worklist.empty()) {
+    auto &[bb, flag]  = worklist.back();
+    if (flag) {
+      worklist.pop_back();
+      map<Value*, Value*> vmap;
+      map<const BasicBlock *, vector<const BasicBlock *>> bbmap;
+      vector<const BasicBlock *> cloned;
+      auto bb_nodes = Nodes[bb];
+
+      for (auto &&bb_ii : bb_nodes) {
+        bbmap[bb_ii].push_back(bb_ii);
+      }
+      for (unsigned unroll = 1 ; unroll <= k; ++unroll) {
+        for (auto &&bb_ii : bb_nodes) {
+          auto &bb_i = const_cast<BasicBlock&>(*bb_ii);
+          string twine = "_" + string(bb_dot_name(bb->getName())) +
```

Comment:
Q: Is `bb_dot_name` needed for correctness?

Reply:
Suggestion: what about using `#` as a prefix for basic blocks that are specially created by Alive2? There is already `#sink` as well.

Reply:
A: Yeah you're right, bb_dot_name is not needed here.

And yeah it is a good suggestion, I could make all cloned BB start with "#".